internal/controller/fluxinstance_artifact_manager.go

```diff
@@ -0,0 +1,35 @@
+// Copyright 2024 Stefan Prodan.
+// SPDX-License-Identifier: AGPL-3.0
+
+package controller
+
+import (
+	"k8s.io/client-go/util/workqueue"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	fluxcdv1 "github.com/controlplaneio-fluxcd/flux-operator/api/v1"
+)
+
+// FluxInstanceArtifactReconcilerOptions contains options for the reconciler.
+type FluxInstanceArtifactReconcilerOptions struct {
+	RateLimiter workqueue.TypedRateLimiter[reconcile.Request]
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *FluxInstanceArtifactReconciler) SetupWithManager(mgr ctrl.Manager, opts FluxInstanceArtifactReconcilerOptions) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&fluxcdv1.FluxInstance{},
```

Comment:
Wouldn't this prevent the main reconciler from ever running due to the FluxInstance being removed from the queue?

Reply:
Maybe this is the reason for the e2e timeout?

Reply:
I'm trying to prepend `Named("flux_instance_artifact").` before the `Watches(...)` to see what happens. In any case, the description of `For(...)` says it is equivalent to the `Watches(...)` I'm using now :cry: :

```
For defines the type of Object being *reconciled*, and configures the ControllerManagedBy to respond to create / delete / update events by *reconciling the object*.

This is the equivalent of calling Watches(source.Kind(cache, &Type{}, &handler.EnqueueRequestForObject{})).
```

Reply:
lol, but it works! the e2e passed locally

Reply:
Looks like `For` creates the name `tolower(Kind)`, so `fluxinstance`. I'm gonna change the name to `fluxinstance_artifact`.